wgpu-hal/src/dx12/mod.rs

```diff
@@ -541,11 +545,23 @@ pub struct Surface {
     target: SurfaceTarget,
     supports_allow_tearing: bool,
     swap_chain: RwLock<Option<SwapChain>>,
+    options: wgt::Dx12BackendOptions,
 }
 
 unsafe impl Send for Surface {}
 unsafe impl Sync for Surface {}
 
+impl Surface {
+    pub unsafe fn wait_for_frame_latency_object(
+        &self,
+        timeout: Option<std::time::Duration>,
+    ) -> Result<bool, crate::SurfaceError> {
```

Comment:
This feels weird being pub

Reply:
My idea was that you don't need to expose the actual waitable handle this way. So `wgpu` can still handle the configuration of the swap chain and creation of the waitable object but we can abstract away the actual unsafe calls syscalls.

Reply:
I think for this particular use case (user demands a certain usage of the waitable object) - it's reasonable to expose the raw handle so they can do whatever they want to do with it (maybe wait for multiple of them at once or something similarly funky.

Reply:
Yes, that's fine! I'll expose the waitable object and make the wait method private.